sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferIdInfoLiveData.java

```diff
@@ -0,0 +1,477 @@
+     */
+    private WorkInfo.State getLastWorkInfoState() {
+        return this.lastWorkInfoState;
+    }
+
+    /**
+     * Store the current {@link WorkInfo.State} so that if needed it can be used while
+     * preparing to send next event to the transfer Observer.
+     *
+     * @param state the WorkInfo state
+     */
+    private void setLastWorkInfoState(WorkInfo.State state) {
+        this.lastWorkInfoState = state;
+    }
+
+    /**
+     * Get the LiveData that stream {@link WorkInfo} of a transfer worker corresponding to the input transfer id.
+     *
+     * This method uses {@link WorkManager} API to retrieve the LiveData of a transfer worker
+     * that is processing the transfer identified by the transfer id emitted by {@code inputTransferIdOrErrorLiveData}.
+     *
+     * @param context the context
+     * @return a LiveData of {@link WorkInfo}
+     */
+    private LiveData<WorkInfo> mapInputTransferIdToWorkInfoLiveData(Context context) {
+        LiveData<List<WorkInfo>> workInfoListLiveData = Transformations
+            .switchMap(
+                this.transferIdOrErrorLiveData,
+                transferIdOrError -> {
+                    setTransferIdOrError(transferIdOrError);
+                    if (this.inputTransferIdOrError.isError()) {
+                        // An error from TransferClient. To continue the LiveData pipeline it is required
+                        // to return non-null LiveData. Null from switchMapFunction will cut the pipeline.
+                        MutableLiveData<List<WorkInfo>> emptyWorkInfoList = new MutableLiveData<>();
+                        emptyWorkInfoList.setValue(null);
+                        return emptyWorkInfoList;
+                    } else {
+                        // No error from TransferClient i.e. transfer work may exists, get the underlying
+                        // LiveData<WorkInfo> for the transfer work.
+                        final long transferId = this.inputTransferIdOrError.getId();
+                        return WorkManager.getInstance(context)
+                            .getWorkInfosForUniqueWorkLiveData(TransferClient.toTransferUniqueWorkName(transferId));
+                    }
+                }
+            );
+        return Transformations.map(workInfoListLiveData, workInfoList -> {
+            if (this.inputTransferIdOrError.isError()) {
+                // An error from TransferClient then emit null WorkInfo. The downstream should check error before
+                // start processing the WorkInfo.
+                return null;
+            } else {
+                final long transferId = this.inputTransferIdOrError.getId();
+                if (workInfoList == null || workInfoList.isEmpty()) {
+                    Log.e(TAG, "Received null or Empty WorkInfo list for the transfer '" + transferId + "' from WorkManager." );
+                    return null;
+                }
+                if (workInfoList.size() > 1) {
```

Comment:
When would we get more than one WorkInfo? When the operation has been paused and resumed multiple times for example? If so, why do we only return the first one? Does that mean they all contain the same info?

Reply:
WorkManager supports chaining of workers and assigning a name to the chain, in such cases this WorkManager API returns a List of WorkInfo. In our case chaining is not used and each transfer/work has a unique name.